tests/unittests/net/test_dhcp.py

```diff
@@ -1199,12 +1199,63 @@ def test_parse_lease_dump(self):
         assert "255.255.240.0" == parsed_lease["subnet-mask"]
         assert "192.168.0.1" == parsed_lease["routers"]
 
+    @pytest.mark.parametrize(
+        "lease, parsed",
+        (
+            pytest.param(
+                """
+                fail
+                """,
+                {},
```

Comment:
I don't think we want this to succeed if the lease doesn't provide any readable lease content. I think it's more appropriate for us to raise an InvalidDHCPLease error or NoDHCPLeaseError in this case to avoid trying to proceed on completely empty/invalid config.